#184752075 Configure Postgres DB and Sequelize ORM

irakevine opened the pull request:
## What does this PR do?
Configure The App to use PostgreSQL as DB Engine and Sequelize ORM

## Description of Task to be completed?
1. Setup Postgres
2. Setup Sequelize ORM
3. Tested DB Connection

## How should this be manually tested?

1. Clone the repo
2. Run `npm install`
3. Create a `.env` file in the root of the app and add all necessary environment variables.
Check `.env.example` to find required environment variables need to run the application.
4. Run `npm run migrate` to create table
5. Run `npm run seed` to create seeders
6. Use postman to test this route `localhost:3000/api/users`. where 3000 is a  port number which can modified accordingly
 
## Any background context you want to provide?
The app comes with support for MONGODB, a NoSQL Database. PostgreSQL is an Open Source, fully featured and optimized SQL database engine for production ready environments.

## What are the relevant pivotal tracker/Trello stories?
1. [#184752075](https://www.pivotaltracker.com/n/projects/2635056/stories/184752075)

bergad37 reviewed (changes requested):
Can we add an undoing migrations script?

nishimweprince reviewed (approved):
The setup is complete. It will make it very easy to add models and perform migrations when we complete the ERD.

nshutijonathan commented on src/server.js:
```diff
@@ -14,10 +15,41 @@ app.use(cors());
-  console.log(error);
-}
+
+// sequelize
```
Remove commented out codes

nshutijonathan commented on src/server.js:
```diff
@@ -14,10 +15,41 @@ app.use(cors());
+  })
+  .catch((err) => {
+    /* eslint-disable */
+    console.error('Unable to connect to the database:', err);
```
I would go with a `return err.message` instead of logging the error

nshutijonathan reviewed (changes requested):
I left some comments

nishimweprince committed:
chore (configuration):postgres configuration and sequelize ORM

-database connection
-installation sequelize
-create dummy user table and model setup to test to test the whole
configuration

Delivers #184752075

nishimweprince commented on database/models/index.js:
```diff
@@ -0,0 +1,43 @@
+
+db.sequelize = sequelize;
+db.Sequelize = Sequelize;
+
```
We need to define a users table using the User model and Sequelize instance defined above before exporting the `db` object.

Something similar to this: `db.users = require(./user.js)(sequelize, Sequelize);`. You may refer to this [link](https://github.com/nishimweprince/atlp-bookstore-bn/blob/main/database/models/index.cjs#L64)

nishimweprince commented on database/models/index.js:
```diff
@@ -0,0 +1,43 @@
+if (config.use_env_variable) {
+  sequelize = new Sequelize(process.env[config.use_env_variable], config);
+} else {
+  sequelize = new Sequelize(config.database, config.username, config.password, config);
```
Since we modified the database configuration file located here: `config/config.cjs`, consider also replacing the Sequelize instance with:

`const sequelize = new Sequelize(
'database', 'username', 'password' {host: 'host', dialect: "postgres"});` -- replacing parameters with your database credentials. 
You may also refer to this [link](https://sequelize.org/docs/v6/getting-started/#connecting-to-a-database)

nishimweprince reviewed (commented):
I added suggestions for your review.

irakevine committed:
chore (configuration): postgres                                                       configuration and sequelize ORM                                                                             -database connection                                                                                      -installation Sequelize                                                                                   -create dummy user table and model setup to test the whole configuration                                delivers #184752075

nshutijonathan reviewed (approved):
LGTM

nishimweprince reviewed (commented):
The configuration is complete.

niyontwali reviewed (approved):
Well done

jcharles921 reviewed (approved):
Nice 👍

nishimweprince reviewed (approved):
Great work!
